src/main/java/io/pravega/connectors/flink/FlinkPravegaReader.java

```diff
@@ -220,6 +258,150 @@ private void triggerCheckpoint(String checkpointIdentifier) throws FlinkExceptio
         checkpointTrigger.triggerCheckpoint(checkpointId);
     }
 
+    // ------------------------------------------------------------------------
+    //  metrics
```

Comment:
I'm not sure if you want to introduce this here, but we could also add `unreadBytes` from the metrics interface. We should be adding more metrics in the future:

https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/stream/ReaderGroupMetrics.java

I'm also wondering if we should have metrics for number of segments out of the listener:

https://github.com/pravega/pravega/tree/master/client/src/main/java/io/pravega/client/stream/notifications